include/swift/AST/KnownIdentifiers.def

```diff
@@ -53,6 +53,7 @@ IDENTIFIER(decode)
 IDENTIFIER(decodeIfPresent)
 IDENTIFIER(Decoder)
 IDENTIFIER(decoder)
+IDENTIFIER_WITH_NAME(Differentiation, "_Differentiation")
```

Comment:
Note that you can use `IDENTIFIER_` for this case (it's not a big deal).

Reply:
Done in https://github.com/apple/swift/pull/27511/commits/5aececee414f9a56adfed3f751a24df2d4a08c5e.

Locally verified that `ASTContext::getProtocol(KnownProtocolKind::Differentiable)` works (returns non-null pointer).